Windows/TestHarnesses/T1543.003_WindowsService/ServiceInstaller.ps1

```diff
@@ -1,740 +1,912 @@
+* LoadCount - The number of times the driver has been loaded in kernel memory. This field may not be populated if the driver is not currently loaded.
+
+.EXAMPLE
+
+New-ATHService -ServiceName phymem -DisplayName 'Does driver stuff' -ServiceType KernelDriver -FilePath phymem64.sys -StartService
+
+.EXAMPLE
+
+New-ATHService -ServiceName TestService -DisplayName TestService -FilePath filename.exe
+
+.EXAMPLE
+
+New-ATHService -ServiceName TestService -DisplayName TestService -FilePath filename.exe -StartService
+
+.EXAMPLE
+
+New-ATHService -ServiceName TestService -DisplayName TestService -FilePath filename.exe -Variant Win32 -StartType DemandStart -StartService
+
+.EXAMPLE
+
+New-ATHService -ServiceName TestService -DisplayName TestService -FilePath filename.exe -Variant sc.exe -StartType AutoStart -ServiceType Win32OwnProcess -StartService
+#>
+
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory)]
+        [String]
+        [ValidateLength(0,256)]
+        $ServiceName,
+
+        [Parameter(Mandatory)]
+        [String]
+        [ValidateLength(0,256)]
+        $DisplayName,
+
+        [string]
+        [ValidateSet('sc.exe', 'WMI', 'Win32', 'Registry')]
+        $Variant = 'sc.exe',
+
+        [String]
+        [ValidateSet('BootStart', 'SystemStart', 'AutoStart', 'DemandStart', 'Disabled')]
+        $StartType = 'DemandStart',
+
+        [String]
+        [ValidateSet('KernelDriver', 'FileSystemDriver', 'Win32OwnProcess', 'Win32ShareProcess')]
+        $ServiceType = 'Win32OwnProcess',
+
+        [Parameter(Mandatory)]
+        [String]
+        [ValidateScript({Test-Path -Path $_ -PathType Leaf})]
+        $FilePath,
+
+        [Switch]
+        $StartService
+    )
+
+    $ServicesRegKey = 'HKLM:\SYSTEM\CurrentControlSet\Services'
+    $NewServiceRegKey = Join-Path -Path $ServicesRegKey -ChildPath $ServiceName
+    $ServiceRegistryKey = $null
+    $TestCommand = $MyInvocation
+
+    # Resolve the full service binary path
+    $ServiceBinPath = Resolve-Path -Path $FilePath -ErrorAction Stop
+
+    Write-Verbose "Requesting service control manager handle with SC_MANAGER_CONNECT access."
+
+    switch($Variant) {
+       'sc.exe'{
+        #Updating values to match: https://learn.microsoft.com/en-us/windows-server/administration/windows-commands/sc-create
+           switch($ServiceType){
+                'KernelDriver' {$service='kernel'}
+                'FileSystemDriver'{$service='filesys'}
+                'Win32OwnProcess'{$service='own'}
+                'Win32ShareProcess'{$service='share'}
+                }
+            switch($StartType){
+                'AutoStart'{$start = 'auto'}
+                'BootStart'{$start = 'boot'}
+                'SystemStart'{$start = 'system'}
+                'DemandStart'{$start = 'demand'}
+                'Disabled'{$start = 'disabled'}
+            }
+
+       $null = sc.exe create $ServiceName binpath= $FilePath start= $start displayname= $DisplayName type= $service
```

Comment:
In other test harnesses, Win32_Process Create is used to spawn processes since it returns a return code and process ID which should be displayed in the object output. Users would ideally want to get process context for sc.exe in the object output.